staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -901,8 +925,28 @@ func compareUpdateCalls(t *testing.T, left, right []fakecloud.UpdateBalancerCall
 	}
 }
 
+// compareHostSets compares if the nodes in left are in right, despite the order.
+func compareHostSets(t *testing.T, left, right []*v1.Node) bool {
+	if len(left) != len(right) {
+		return false
+	}
+	for _, lHost := range left {
```

Comment:
for followup?

staging/src/k8s.io/apimachinery/pkg/util/sets:

```
l := sets.New(left...)
r := sets.New(right...)
return reflect.DeepEqual(l, r)
```

?

Reply:
OK, this comment made me go down a two hour rabbithole into the depths of client-go, and I still haven't come out sane. At first I thought: "makes total sense, let me update the code" - except the test started failing, at which point I started my journey. 

By doing this: `reflect.DeepEqual` will start comparing the pointers of each item in the `set`s (read: `map`s) - and these differ. That is to say: the objects/pointers of `node1`/`node2` as retrieved in the informer cache are not the objects/pointers as the test defines at the top, here: https://github.com/kubernetes/kubernetes/blob/b8c22aa0e2b565b98daafdf6f1b498c5e4385f48/staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go#L1398-L1399

At one point (I don't know where...) something must be doing a `.DeepCopy` on these objects when populating the cache.

Given that the current state of the code does a `reflect.DeepEqual(nodeLeft, nodeRight)` the comparison changes and underlying values of the struct fields are compared (see: https://github.com/golang/go/blob/master/src/reflect/deepequal.go#L123-L127) - so the test passes. 

My question is: does this mean the test setup is wrong, or rather that this comment doesn't really work?